### PR TITLE
2 process data

### PR DESCRIPTION
**Pull request by Ukaypromise**
### Hello Reviewer 👋🏾 ,
**For this PR the following requirements where ensured:**

- **Create model classes for all entities as shown in the ERD diagram.**
- **Set up associations between models.**
- **Create custom methods**

1. A method that returns the 3 most recent posts for a given user.
2. A method that updates the posts counter for a user.
3. A method which returns the 5 most recent comments for a given post.
4. A method that updates the comments counter for a post.
5. A method that updates the likes counter for a post.

**Thanks in anticipation 👍🏾**

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  after_save :update_post_counter
+
+  def five_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_post_counter
+    author.update(posts_counter: author.posts.all.length)
+  end
```
Comment (codepantha) [inline app/models/post.rb]: - [x] [OPTIONAL] Good approach to solving this problem. I'd like to suggest a shorter and even more readable way to do this by using the `increment` method as shown below 👇 
``` ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+  after_save :update_comment_counter
+
+  def update_comment_counter
+    post.update(comments_counter: post.comments.all.length)
+  end
```
Comment (codepantha) [inline app/models/comment.rb]: - [x] [OPTIONAL] Please consider using the `increment` method I talked about earlier in order to make your code more readable and concise 👍 
``` ruby
def update_comment_counter
  post.increment!(comments_counter)
end
```

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_save :update_like_counter
+
+  def update_like_counter
+    post.update(like_counter: post.likes.all.length)
+  end
```
Comment (codepantha) [inline app/models/like.rb]: - [x] [OPTIONAL] Good approach to solving this problem. I'd like to suggest a shorter and even more readable way to do this by using the increment method as shown below 👇
``` ruby
def update_like_counter
  post.increment!(:likes_counter)
end
```

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED ✔️ 🟢 

## Hi @Ukaypromise 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR 📌 
✔️ Good commit history 📌 
✔️ Professional README 📌 
✔️ No linter errors 📌 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] Please consider fixing the link to your LICENSE file in the README 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (Ukaypromise) [reply app/models/post.rb]: Thank you namesake @codepantha 
I have implemented all optional changes.

Comment (codepantha) [reply app/models/post.rb]: Yeah, you stole my name 🤣. And you're welcome 👍

**Pull request merged**